teachertool/src/transforms/loadReadableBlockName.ts

```diff
@@ -0,0 +1,18 @@
+import { getBlockAsText } from "../services/makecodeEditorService";
+import { cacheBlockAsText, getCachedBlockAsText } from "../services/storageService";
+
+export async function loadBlockAsText(blockId: string): Promise<pxt.editor.BlockAsText | undefined> {
+    // Check for cached version.
+    let readableBlockName = getCachedBlockAsText(blockId);
+    if (readableBlockName) {
+        return Promise.resolve(readableBlockName);
```

Comment:
Is this `Promise.resolve` needed here? If a truthy value is returned from this function, doesn't that mean the promise has resolved?

Reply:
async functions don't need Promise.resolve like this, yeah